internal/pkg/v2/infrastructure/redis/client.go

```diff
@@ -24,15 +25,13 @@ type Client struct {
 	*redisClient.Client
 }
 
-func NewClient(config db.Configuration, logger logger.LoggingClient) (*Client, error) {
-	var err error
-	dc := &Client{}
+func NewClient(config db.Configuration, logger logger.LoggingClient) (dc *Client, err error) {
```

Comment:
Why not returning v2.EdgexError like in event.go?

Reply:
you are right. we should use v2.EdgexError here

Reply:
modified